tests/test_data.py

```diff
@@ -196,3 +198,198 @@ def test_zero_length_fails():
         types.Transitions(
             obs=empty, acts=empty, next_obs=empty, dones=empty.astype(np.bool),
         )
+
+
+DICT_DATASET_PARAMS = [
+    (dataset.EpochOrderDictDataset, {"shuffle": False}),
+    (dataset.EpochOrderDictDataset, {"shuffle": True}),
+    (dataset.RandomDictDataset, {}),
+]
+
+DATA_MAP = [
+    {"a": np.zeros([10, 2], dtype=int), "b": np.random.standard_normal([10])},
+    {"asdf": np.zeros([100, 2, 4])},
+    {"asdf": np.zeros([97, 2, 4]), "foo": np.ones([97, 1], dtype=bool)},
+]
+
+
+@pytest.fixture(params=DICT_DATASET_PARAMS)
+def dict_dataset_params(request):
+    return request.param
+
+
+@pytest.fixture(params=DATA_MAP)
+def data_map(request):
+    return copy.deepcopy(request.param)
+
+
+@pytest.fixture
+def dict_dataset(dict_dataset_params, data_map):
+    dataset_cls, kwargs = dict_dataset_params
+    return dataset_cls(data_map, **kwargs)
+
+
+def test_dict_dataset_copy_data_map(dict_dataset: dataset.DictDataset, data_map):
+    """Check that `dict_dataset.data_map` is unchanged by writes to `data_map` param.
+
+    Note that the `data_map` fixture supplies the same instance `data_map` that was used
+    to initialize the `dict_dataset` fixture.
+    """
+    backup_map = copy.deepcopy(dict_dataset.data_map)
+    npt.assert_equal(backup_map, dict_dataset.data_map)
+    for v in data_map.values():
+        # Multiply array in-place by 2.
+        np.multiply(v, v.dtype.type(2), out=v)
```

Comment:
Most of your arrays are all-zero... so multiplying by two is a no-op.

Reply:
I'll try adding, too. I think I used multiplying as a second choice because I was running into some dtype issues with the `bool` array.